src/device.rs

```diff
@@ -1136,7 +1136,38 @@ impl<E: StreamSample> TxStream<E> {
         Ok(())
     }
 
-    // TODO: read_status
+    /// Read the status of the stream.
+    /// 
+    /// This is required to detect underflows and such as they are not reported by 
+    /// [write](TxStream::write).
+    /// 
+    /// `chan_mask``, `flags``, and `time_ns`` are output parameters and _may_ be 
+    /// set depending on the type of status result.
+    /// 
+    /// Returns the status `Result`, usually this will be an [Error], as would be 
+    /// returned from a stream read/write.
+    /// 
+    /// [ErrorCode::Timeout] should generally be ignored.
+    /// 
+    /// Note that `timeout_us` is only `i32` on Windows and panics if the value is 
+    /// too large for `i32` on that platform.
+    pub fn read_status(&mut self, chan_mask: &mut usize, flags: &mut i32, time_ns: &mut i64, timeout_us: i64) -> Result<usize, Error> {
+        // Conversion needed for Windows, which takes an i32 here for some reason.
```

Comment:
It's defined as `long` in the C header, which is only 32 bits on 64-bit Windows.